newrelic/resource_newrelic_alert_condition_test.go

```diff
@@ -45,126 +53,148 @@ func TestAccNewRelicAlertCondition_Basic(t *testing.T) {
+				ExpectNonEmptyPlan: false,
+			},
+			// Test: Update
+			{
+				Config: testAccCheckNewRelicAlertConditionConfigUpdated(rNameUpdated),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckNewRelicAlertConditionExists("newrelic_alert_condition.foo"),
 					resource.TestCheckResourceAttr(
-						"newrelic_alert_condition.foo", "name", fmt.Sprintf("tf-test-updated-%s", rName)),
+						"newrelic_alert_condition.foo", "name", rNameUpdated),
 					resource.TestCheckResourceAttr(
-						"newrelic_alert_condition.foo", "enabled", "true"),
+						"newrelic_alert_condition.foo", "type", "apm_app_metric"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_condition.foo", "enabled", "false"),
 					resource.TestCheckResourceAttr(
 						"newrelic_alert_condition.foo", "runbook_url", "https://bar.example.com"),
 					resource.TestCheckResourceAttr(
 						"newrelic_alert_condition.foo", "entities.#", "1"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_condition.foo", "metric", "error_percentage"),
+					resource.TestCheckResourceAttr(
+						"newrelic_alert_condition.foo", "condition_scope", "application"),
 					resource.TestCheckResourceAttr(
 						"newrelic_alert_condition.foo", "term.#", "1"),
 					resource.TestCheckResourceAttr(
-						"newrelic_alert_condition.foo", "term.1944209821.duration", "10"),
+						"newrelic_alert_condition.foo", "term.3409672004.duration", "10"),
 					resource.TestCheckResourceAttr(
-						"newrelic_alert_condition.foo", "term.1944209821.operator", "below"),
+						"newrelic_alert_condition.foo", "term.3409672004.operator", "above"),
 					resource.TestCheckResourceAttr(
-						"newrelic_alert_condition.foo", "term.1944209821.priority", "critical"),
+						"newrelic_alert_condition.foo", "term.3409672004.priority", "critical"),
 					resource.TestCheckResourceAttr(
-						"newrelic_alert_condition.foo", "term.1944209821.threshold", "0.65"),
+						"newrelic_alert_condition.foo", "term.3409672004.threshold", "1"),
 					resource.TestCheckResourceAttr(
-						"newrelic_alert_condition.foo", "term.1944209821.time_function", "all"),
+						"newrelic_alert_condition.foo", "term.3409672004.time_function", "any"),
 				),
 			},
+			// Test: Import
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
 		},
 	})
 }
 
 func TestAccNewRelicAlertCondition_ZeroThreshold(t *testing.T) {
-	rName := acctest.RandString(5)
-	resource.Test(t, resource.TestCase{
+	rName := fmt.Sprintf("tf-test-%s", acctest.RandString(5))
+	resource.ParallelTest(t, resource.TestCase{
 		PreCheck:     func() { testAccPreCheck(t) },
 		Providers:    testAccProviders,
 		CheckDestroy: testAccCheckNewRelicAlertConditionDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccCheckNewRelicAlertConditionConfigZeroThreshold(rName),
+				Config: testAccCheckNewRelicAlertConditionConfigThreshold(rName, 0),
```

Comment:
Not a blocker, but I noticed a pattern in testing examples that the `Config` functions didn't have the `Check` word as part of it. And the subsequent functions passed to `ComposeTestCheckFunc()` contained the `Check` word since it's doing a check vs returning a config.

Reply:
Here's an [example](https://www.terraform.io/docs/extend/testing/acceptance-tests/testcase.html) that I was going off of.

Reply:
I will update - I had meant to put this in.